### PR TITLE
Dockerize

### LINKED ISSUE #24: Dockerise the repository

### PR DESCRIPTION
**Pull request by pterodragon**
Closes #24.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        links: 
+          - mysql
+          - redis
+        depends_on:
+          - mysql
+          - redis
+        ports:
+          - '3000:3000'
+          - '3001:3001'
+        command: bash -c 'cd server && while ! (mysqladmin ping --host=mysql --user=algo-dev --password=development > /dev/null 2>&1); do echo "mysql not yet ready"; sleep 3; done; yarn prepare-db; yarn start'
```
Comment (samueltangz) [inline docker-compose.yml:36]: are there any ways to avoid entering username and password through command line?

File: .circleci/config.yml [hunk 2]
```diff
@@ -59,6 +59,25 @@ jobs:
       - run:
           name: Running server tests
           command: cd server && yarn test
+  main:
```
Comment (samueltangz) [inline .circleci/config.yml:62]: the job `main` may not be as descriptive as what it does

File: .circleci/config.yml [hunk 3]
```diff
@@ -69,4 +88,5 @@ workflows:
           - prepare
       - test_server:
           requires:
-          - prepare
\ No newline at end of file
+          - prepare
+      - main
```
Comment (samueltangz) [inline .circleci/config.yml:92]: would it require anything beforehand for it to run?

File: docker-compose.yml (hunk 1 above)
Comment (pterodragon) [reply docker-compose.yml:36]: Should read from env vars or sth

File: .circleci/config.yml (hunk 3 above)
Comment (pterodragon) [reply .circleci/config.yml:92]: no, it uses a circleci base image having the necessary tools like `docker-compose` and directly builds the docker image

File: .circleci/config.yml (hunk 2 above)
Comment (pterodragon) [reply .circleci/config.yml:62]: Suggestion?

Comment (samueltangz) [reply .circleci/config.yml:62]: `test_serve`?

**Pull request merged**